### ISSUE TITLE
auto-value Error: Nullable is not a repeatable annotation type

### ISSUE THREAD

**Issue by natros** [google/auto#355]
This code is falling with Error:(29, 3) java: org.jetbrains.annotations.Nullable is not a repeatable annotation type

``` java
import com.google.auto.value.AutoValue;
import org.jetbrains.annotations.Nullable;

@AutoValue
public abstract class User {
    abstract Integer userId();
    @Nullable abstract String login();
    abstract String name();
    public static Builder builder() {
        return new AutoValue_User.Builder();
    }

    @AutoValue.Builder
    public abstract static class Builder {
        abstract Builder userId(Integer userId);
        abstract Builder login(String login);
        abstract Builder name(String name);
        abstract User build();
    }
}
```

And the generated code

``` java
import javax.annotation.Generated;
import org.jetbrains.annotations.Nullable;

@Generated("com.google.auto.value.processor.AutoValueProcessor")
 final class AutoValue_User extends User {

  private final Integer userId;
  private final String login;
  private final String name;

  private AutoValue_User(
      Integer userId,
      @Nullable String login,
      String name) {
    this.userId = userId;
    this.login = login;
    this.name = name;
  }

  @Override
  Integer userId() {
    return userId;
  }

  @Nullable
  @Nullable
  @Override
  String login() {
    return login;
  }

  @Override
  String name() {
    return name;
  }

  @Override
  public String toString() {
    return "User{"
        + "userId=" + userId + ", "
        + "login=" + login + ", "
        + "name=" + name
        + "}";
  }

  @Override
  public boolean equals(Object o) {
    if (o == this) {
      return true;
    }
    if (o instanceof User) {
      User that = (User) o;
      return (this.userId.equals(that.userId()))
           && ((this.login == null) ? (that.login() == null) : this.login.equals(that.login()))
           && (this.name.equals(that.name()));
    }
    return false;
  }

  @Override
  public int hashCode() {
    int h = 1;
    h *= 1000003;
    h ^= this.userId.hashCode();
    h *= 1000003;
    h ^= (login == null) ? 0 : this.login.hashCode();
    h *= 1000003;
    h ^= this.name.hashCode();
    return h;
  }

  static final class Builder extends User.Builder {
    private Integer userId;
    private String login;
    private String name;
    Builder() {
    }
    Builder(User source) {
      this.userId = source.userId();
      this.login = source.login();
      this.name = source.name();
    }
    @Override
    public User.Builder userId(Integer userId) {
      this.userId = userId;
      return this;
    }
    @Override
    public User.Builder login(@Nullable String login) {
      this.login = login;
      return this;
    }
    @Override
    public User.Builder name(String name) {
      this.name = name;
      return this;
    }
    @Override
    public User build() {
      String missing = "";
      if (userId == null) {
        missing += " userId";
      }
      if (name == null) {
        missing += " name";
      }
      if (!missing.isEmpty()) {
        throw new IllegalStateException("Missing required properties:" + missing);
      }
      return new AutoValue_User(
          this.userId,
          this.login,
          this.name);
    }
  }

}
```

I'm using  auto-value 1.3-rc2 with java 8

**Comment by eamonnmcmanus** [comment]
That seems surprising. I'm not sure how it's different from cases like [this test](https://github.com/google/auto/blob/master/value/src/it/functional/src/test/java/com/google/auto/value/AutoValueTest.java#L463) which passes.

**Comment by natros** [comment]
This only happens for org.jetbrains.annotations.Nullable annotation. For javax.annotation.Nullable there is no error.

**Comment by eamonnmcmanus** [comment]
I think the issue is [here](https://github.com/google/auto/blob/master/value/src/main/java/com/google/auto/value/processor/AutoValueProcessor.java#L220). I'm thinking that in some circumstances both for-loops might contribute the same annotation to the list. Perhaps this happens when an annotation is both `TYPE` and `TYPE_USE`? @brychcy what do you think?

**Comment by brychcy** [comment]
Ugh. You are right. They really have created a "mixed-up" Null annotation:

https://github.com/JetBrains/intellij-community/blob/master/platform/annotations/java8/src/org/jetbrains/annotations/Nullable.java

This is a bad idea because of the way they work with arrays, e.g.:
@Nullable String[] someField;
which with such a @Nullable means that it is a nullable field that contains an array of nullable Strings, i.e. the @Nullable applies both to the array element type and the field. Now think of combinations with @NotNull. Maybe they did it because Intellij used to have a bug where the positioning of TYPE_USE null annotations w.r.t. arrays was interpreted wrong (https://youtrack.jetbrains.com/issue/IDEA-139253)

In general, I would recommend against using such (conceptually broken) null annotations, but actually this is not a issue for Autovalue as it supports only primitive arrays anyway (and as I wrote in the patch, the Nullable would currently not be output for those in the right position).

If you think these kind of annotations should be supported by Autovalue, a possible fix would be to do a duplicate detection in the place you marked.

**Comment by brychcy** [comment]
see https://github.com/google/auto/pull/356

**Comment by trevjonez** [comment]
A work around that may be acceptable in **_some_** cases is to use `org.jetbrains:annotations-java5:15.0` instead of `org.jetbrains:annotations:15.0`

**Comment by eamonnmcmanus** [comment]
This should be fixed in 1.4-SNAPSHOT. [This test](https://github.com/google/auto/blob/master/value/src/it/functional/src/test/java/com/google/auto/value/AutoValueJava8Test.java#L494) covers the case discussed above.
